### ISSUE TITLE
What is the loss function?

### ISSUE THREAD

**Issue by yuye2133** [dhwajraj/deep-siamese-text-similarity#8]
I cannot understand the loss function, could you explain it for me

**Comment by dhwajraj** [comment]
Please refer to HADSELL, R., CHOPRA, S., AND LECUN, Y. 2006. _Dimensionality reduction by learning an invariant mapping_. In CVPR, IEEE Press.

**Comment by 1000sprites** [comment]
self.distance = tf.sqrt(tf.reduce_sum(tf.square(tf.subtract(self.out1, self.out2)), 1, keep_dims=True))
self.distance = tf.div(self.distance, tf.add(tf.sqrt(tf.reduce_sum(tf.square(self.out1), 1, keep_dims=True)),tf.sqrt(tf.reduce_sum(tf.square(self.out2),1,keep_dims=True))))
the first self.distance is Euclidean Distance, what is the second self.distance?

**Comment by Huijun-Cui** [comment]
this code really waste my time, why it recommend to me on top page ?   ****
